### PR TITLE
Added tests for `ReceiptFetcher.receiptData` on watchOS and moved to `StoreKitWorkarounds`

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes [CSDK-24].
Fix added in #1625.

[CSDK-24]: https://revenuecats.atlassian.net/browse/CSDK-24?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Tests/UnitTests/Purchasing/ReceiptFetcherTests.swift [hunk 1]
```diff
@@ -136,4 +137,70 @@ class ReceiptFetcherTests: TestCase {
...
 
+    // MARK: - Receipt URL
+
+    func testNoReceiptURLIfBundleDoesNotHaveOne() {
+        self.mockBundle.receiptURLResult = .nilURL
+
+        expect(self.receiptFetcher.receiptURL).to(beNil())
+    }
+
+    func testReceiptURLIsUnchangedInSandboxOnOlderVersionsIfNotWatchOS() throws {
```
Comment (NachoSoto) [inline Tests/UnitTests/Purchasing/ReceiptFetcherTests.swift:148]: This test ensures that the logic is only for watchOS.

**Review by taquitos — APPROVED** [review]
🎈🐐

**Comment by NachoSoto** [comment]
I'll look at the failures tomorrow.

File: Sources/Purchasing/ReceiptFetcher.swift [hunk 2]
```diff
@@ -53,33 +53,50 @@ class ReceiptFetcher {
...
         }
 
         #if os(watchOS)
+        receiptURL = self.watchOSReceiptURL(receiptURL)
+        #endif
+
+        return receiptURL
+    }
+
+    func watchOSReceiptURL(_ receiptURL: URL) -> URL? {
```
Comment (aboedo) [inline Sources/Purchasing/ReceiptFetcher.swift:71]: [not for this PR] I wonder if it'd make sense to move this one to `StoreKitWorkarounds`

**Review by aboedo — APPROVED** [review]
Looks awesome!! 🚢

File: Sources/Purchasing/ReceiptFetcher.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (NachoSoto) [reply Sources/Purchasing/ReceiptFetcher.swift:71]: Smart of me to not compile this watchOS fix on watchOS 🤦🏻

Comment (NachoSoto) [reply Sources/Purchasing/ReceiptFetcher.swift:71]: Oh yeah! Let me do that

Comment (NachoSoto) [reply Sources/Purchasing/ReceiptFetcher.swift]: Done

[the remark this answers is not in the document]
Comment (taquitos) [reply Sources/Purchasing/ReceiptFetcher.swift]: Oh dang, good catch!

**Pull request merged**